### PR TITLE
RAIL-1842 DateFilter Mark available granularities as deprecated

### PR DESCRIPTION
**Pull request by plavemar**
Description of changes.

- RAIL-1842 availableGranularities in DateFilter marked as deprecated
- RAIL-1603 issue with reopening 'to' field after choosing its value in relativeFilterForm
- RAIL-1624 consider available granularities in relativeFilterForm and presets
- RAIL-1872 console warning about incorrect DateFilter setup 

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [x] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [x] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [x] `check` passes
-   [x] `check-extended` passes
-   [x] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by plavemar** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3243/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4899/ : FAILURE in 6m 05s

**Comment by plavemar** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3246/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4902/ : SUCCESS in 20m 18s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/823/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3248/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/823/ : SUCCESS in 26m 42s

**Comment by plavemar** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR1264

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3253/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/364/ : SUCCESS in 9m 09s (non-voting)

**Comment by no23reason** [comment]
Please run `rush change` and mention the deprecation of the `availableGranularities` and what to use instead. This will be mentioned in teh changelog for the next version released.

File: libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx [hunk 1]
```diff
@@ -123,10 +124,28 @@ export class DateFilter extends React.PureComponent<IDateFilterProps, IDateFilte
...
 
+    private static checkInitialFilterOption = (filterOption: DateFilterOption) => {
+        if (isAbsoluteDateFilterForm(filterOption) && !(filterOption.from && filterOption.to)) {
+            // eslint-disable-next-line no-console
+            console.warn(
+                "The default filter option is not valid. Values 'from' and 'to' from absoluteForm filter option must be specified.",
+            );
+        }
+
+        if (isRelativeDateFilterForm(filterOption) && !(filterOption.from && filterOption.to)) {
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx:135]: This is not correct, `from: 0` and `to: 0` are valid values and they should not trigger the warning. You should probably use `isNil` from lodash here.

File: libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx [hunk 2]
```diff
@@ -123,10 +124,28 @@ export class DateFilter extends React.PureComponent<IDateFilterProps, IDateFilte
...
+                "The default filter option is not valid. Values 'from' and 'to' from relativeForm filter option must be specified.",
+            );
+        }
+    };
+
     constructor(props: IDateFilterProps) {
         super(props);
         this.state = DateFilter.getStateFromProps(props);
 
+        DateFilter.checkInitialFilterOption(props.selectedFilterOption);
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx:147]: Please put this in `componentDidMount` as that is the place for side-effects, not constructor

Comment (plavemar) [reply libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx]: Function call moved to `componentDidMount`

File: libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx (hunk 1 above)
Comment (plavemar) [reply libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx]: isNil reviewed and used to allow '0' as a valid default filter value

**Comment by plavemar** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3257/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4913/ : SUCCESS in 15m 07s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/827/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3259/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/827/ : SUCCESS in 20m 20s

**Review by no23reason — APPROVED** [review]
LGTM

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1264

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/980/ : SUCCESS in 1m 14s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1250/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4918/ : SUCCESS in 12m 39s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/368/ : SUCCESS in 10m 15s (non-voting)

**Pull request merged**